src/common/function/src/scalars/timestamp/to_unixtime.rs

```diff
@@ -37,17 +32,23 @@ const NAME: &str = "to_unixtime";
 
 fn convert_to_seconds(arg: &str) -> Option<i64> {
     match Timestamp::from_str(arg) {
-        Ok(ts) => {
-            let sec_mul = (TimeUnit::Second.factor() / ts.unit().factor()) as i64;
-            Some(ts.value().div_euclid(sec_mul))
-        }
-        Err(_err) => None,
+        Ok(ts) => Some(ts.split().0),
+        Err(_err) => match Date::from_str(arg) {
```

Comment:
Maybe we also need to consider `DateTime` string literals?

Reply:
Make sense

Reply:
Already supported